hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/utils/EntityIdUtil.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    public static Address asTypedEvmAddress(final AccountID id) {
+        return Address.wrap(Bytes.wrap(asEvmAddress(id)));
+    }
+
+    public static Address asTypedEvmAddress(final ContractID id) {
+        return Address.wrap(Bytes.wrap(asEvmAddress(id)));
+    }
+
+    public static Address asTypedEvmAddress(final TokenID id) {
+        return Address.wrap(Bytes.wrap(asEvmAddress(id)));
+    }
+
+    public static byte[] asEvmAddress(final int shard, final long realm, final long num) {
+        final byte[] evmAddress = new byte[20];
+
+        arraycopy(Ints.toByteArray(shard), 0, evmAddress, 0, 4);
+        arraycopy(Longs.toByteArray(realm), 0, evmAddress, 4, 8);
+        arraycopy(Longs.toByteArray(num), 0, evmAddress, 12, 8);
+
+        return evmAddress;
+    }
+
+    public static long shardFromEvmAddress(final byte[] bytes) {
+        return Ints.fromByteArray(Arrays.copyOfRange(bytes, 0, 4));
+    }
+
+    public static long realmFromEvmAddress(final byte[] bytes) {
+        return Longs.fromByteArray(Arrays.copyOfRange(bytes, 4, 12));
+    }
+
+    public static long numFromEvmAddress(final byte[] bytes) {
+        return Longs.fromBytes(
+                bytes[12], bytes[13], bytes[14], bytes[15], bytes[16], bytes[17], bytes[18],
+                bytes[19]);
+    }
+
+    public static AccountID accountIdFromEvmAddress(final Address address) {
+        return accountIdFromEvmAddress(address.toArrayUnsafe());
+    }
+
+    public static ContractID contractIdFromEvmAddress(final Address address) {
+        return contractIdFromEvmAddress(address.toArrayUnsafe());
+    }
+
+    public static TokenID tokenIdFromEvmAddress(final Address address) {
+        return tokenIdFromEvmAddress(address.toArrayUnsafe());
+    }
+
+    public static AccountID accountIdFromEvmAddress(final byte[] bytes) {
+        return AccountID.newBuilder()
+                .setShardNum(shardFromEvmAddress(bytes))
+                .setRealmNum(realmFromEvmAddress(bytes))
+                .setAccountNum(numFromEvmAddress(bytes))
+                .build();
+    }
+
+    public static ContractID contractIdFromEvmAddress(final byte[] bytes) {
+        return ContractID.newBuilder()
+                .setShardNum(Ints.fromByteArray(Arrays.copyOfRange(bytes, 0, 4)))
+                .setRealmNum(Longs.fromByteArray(Arrays.copyOfRange(bytes, 4, 12)))
+                .setContractNum(Longs.fromByteArray(Arrays.copyOfRange(bytes, 12, 20)))
+                .build();
+    }
+
+    public static TokenID tokenIdFromEvmAddress(final byte[] bytes) {
+        return TokenID.newBuilder()
+                .setShardNum(Ints.fromByteArray(Arrays.copyOfRange(bytes, 0, 4)))
+                .setRealmNum(Longs.fromByteArray(Arrays.copyOfRange(bytes, 4, 12)))
+                .setTokenNum(Longs.fromByteArray(Arrays.copyOfRange(bytes, 12, 20)))
+                .build();
+    }
+
+    public static Address addressFromBytes(final byte[] bytes) {
```

Comment:
I believe we should get rid of this class and probably keep only this method if needed and used.